midend/parserUnroll.cpp

```diff
@@ -367,6 +366,38 @@ class ParserSymbolicInterpreter {
             auto mc = sord->to<IR::MethodCallStatement>();
             auto e = ev.evaluate(mc->methodCall, false);
             success = reportIfError(state, e);
+        } else if (auto bs = sord->to<IR::BlockStatement>()) {
+            IR::IndexedVector<IR::StatOrDecl> newComponents;
+            for (auto* component : bs->components) {
+                auto newComponent = executeStatement(state, component, valueMap);
+                if (!newComponent)
+                    success = false;
+                else
+                    newComponents.push_back(newComponent);    
+            }
+            sord = new IR::BlockStatement(newComponents);
+        } else if (sord->is<IR::IfStatement>()) {
+            auto ifs = sord->to<IR::IfStatement>()->clone();
+            auto ifcond = ev.evaluate(ifs->condition, true);
+            success = reportIfError(state, ifcond);
+            if (success) {
+                const IR::Statement* ifComponent = nullptr;
+                if (ifcond)
```

Comment:
I don't understand what happens here. ifcond is a pointer to a SymbolicValue.
Why can it be nullptr?

Reply:
There is a check for ifcond value is true or false

Reply:
I think this is wrong. You are comparing a pointer with nullptr here.

Reply:
It should not be possible that ifcond is a nullptr because of the BUG_CHECK in the `get` function. Unless there is some side effect in reportIfError. Ideally, ifCond should be const.

Reply:
Ifcond cannot be nullptr as fruffy already said

Reply:
Yes, that is why the check should not be necessary.

Reply:
Ok, I`ll do it